analyzers/tests/SonarAnalyzer.UnitTest/Common/AnalyzerConfigurationTests.cs

```diff
@@ -32,21 +34,88 @@ namespace SonarAnalyzer.UnitTest.Common
     public class AnalyzerConfigurationTests
     {
         [TestMethod]
-        public void GivenSonarLintXmlIsLoaded_WhenInitializeIsCalledWithDifferentPath_EnabledRulesAreUpdated()
+        public void HotspotConfiguration_WhenInitializeIsCalledWithDifferentPathAndDifferentLoaders_EnabledRulesAreUpdated()
         {
-            var sut = AnalyzerConfiguration.Hotspot;
-            sut.Initialize(new AnalyzerOptions(GetAdditionalFiles(@"a\SonarLint.xml")), GetRuleLoader("S1067"));
+            var sut = new AnalyzerConfiguration.HotspotConfiguration();
+
+            sut.Initialize(new AnalyzerOptions(GetAdditionalFiles(@"bar\SonarLint.xml")), GetRuleLoader("S1067"));
             Assert.IsTrue(sut.IsEnabled("S1067"));
+            Assert.IsFalse(sut.IsEnabled("S9999"));
 
-            sut.Initialize(new AnalyzerOptions(GetAdditionalFiles(@"b\SonarLint.xml")), GetRuleLoader("S1068"));
+            sut.Initialize(new AnalyzerOptions(GetAdditionalFiles(@"qix\SonarLint.xml")), GetRuleLoader("S9999"));
             Assert.IsFalse(sut.IsEnabled("S1067"));
-            Assert.IsTrue(sut.IsEnabled("S1068"));
+            Assert.IsTrue(sut.IsEnabled("S9999"));
+        }
+
+        [TestMethod]
+        public void HotspotConfiguration_WhenInitializeIsCalledWithTheSamePathAndDifferentLoaders_EnabledRulesAreNotUpdated()
+        {
+            var sut = new AnalyzerConfiguration.HotspotConfiguration();
+            var path = "SonarLint.xml";
+
+            sut.Initialize(new AnalyzerOptions(GetAdditionalFiles(path)), GetRuleLoader("S1067"));
+            Assert.IsTrue(sut.IsEnabled("S1067"));
+            Assert.IsFalse(sut.IsEnabled("S9999"));
+
+            sut.Initialize(new AnalyzerOptions(GetAdditionalFiles(path)), GetRuleLoader("S9999"));
+            Assert.IsTrue(sut.IsEnabled("S1067"));
+            Assert.IsFalse(sut.IsEnabled("S9999"));
+        }
+
+        [TestMethod]
+        public void HotspotConfiguration_WhenInitializeIsSecondTimeWithDifferentFileName_EnabledRulesAreNotUpdated()
```

Comment:
Isn't this test identical with `HotspotConfiguration_WhenInitializeIsCalledWithDifferentPathAndDifferentLoaders_EnabledRulesAreUpdated`?

Reply:
no... one calls `Initialize` with SonarLint.xml files, and one with SonarLint and then Foo.xml

I'll try to improve the test names, and maybe refactor to improve readability